File: Solution/0200.cpp
class Solution {
public:
    void dfs_alg(vector<vector<char>>& grid, int i, int j, int m, int n){
        
        if(i >= m || j >= n || j < 0 || i < 0 ||grid[i][j] == '0')
            return;
        
        else{
            grid[i][j] = '0';
            dfs_alg(grid, i-1, j, m, n);
            dfs_alg(grid, i+1, j, m, n);
            dfs_alg(grid, i, j-1, m, n);
            dfs_alg(grid, i, j+1, m ,n);
        }
        
    }
    
    int numIslands(vector<vector<char>>& grid) {
        int m = grid.size(), n = grid[0].size();
        
        int numIsland = 0;
        for(int i=0 ; i < m ; ++i){
            for(int j=0 ; j < n ; ++j){
                if(grid[i][j] == '1'){
                    numIsland += 1;
                    dfs_alg(grid, i, j, m, n);
                }
                
            }
        }
        
        return numIsland;
    }
};

File: Solution/0200.py
class Solution:
    def DFS_alg(self, i, j):
        if i < 0 or i >= len(self.grid):
            return False
        if j < 0 or j >= len(self.grid[0]):
            return False

        if self.grid[i][j] == "1":
            self.grid[i][j] = "0"
            self.DFS_alg(i + 1, j)
            self.DFS_alg(i , j + 1)
            self.DFS_alg(i - 1, j)
            self.DFS_alg(i, j - 1)
        else:
            return False
        

    def numIslands(self, grid: List[List[str]]) -> int:
        self.grid = grid

        count = 0
        for m in range(len(self.grid)):
            for n in range(len(self.grid[0])):
                if self.grid[m][n] == "1":
                    count += 1
                    self.DFS_alg(m, n)
        return count
